### PR TITLE
Add Example to Prospective Parachains Implementers Guide

### PR DESCRIPTION
**Pull request by BradleyOlson64**
After reading the prospective parachains implementers guide I still needed to read a lot of code to conceptually grasp how a fragment tree is built and discarded. This example should help readers grasp those details more easily.

I expect there may be parts phrased poorly or where I'm just wrong 😄

### CODE REVIEW

**Review by slumber — COMMENTED** [review]
Personally, I think this is just going way too far. Even though I'm familiar with fragment trees, it took me a while to get through all notations and examples just to understand what's going on here.

There's a scheme-example of fragment tree above and looks sufficient to me.

I'm not strictly opposing changes, but in my opinion this section of a guide might get ridiculously complicated.

**Comment by BradleyOlson64** [comment]
@slumber I see what you mean. I'll try something simpler, like stating the rules I wanted to highlight in plain english rather than building them all into one example.
